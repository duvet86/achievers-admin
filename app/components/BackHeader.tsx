import { Link } from "@remix-run/react";

import { ArrowLeft } from "iconoir-react";

interface Props {
  to?: string;
}

export function BackHeader({ to = "../" }: Props) {
  return (
    <>
      <div>
        <Link to={to} relative="path" className="btn btn-ghost mb-2 gap-2">
          <ArrowLeft className="w-6" />
          Back
        </Link>
      </div>

      <hr className="mb-4" />
    </>
  );
}
