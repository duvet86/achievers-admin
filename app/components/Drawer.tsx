import { NavLink } from "@remix-run/react";

import { NavArrowRight, User, GraduationCap, Home } from "iconoir-react";

interface Props {
  isAdmin: boolean;
  version: string;
}

interface DrawerLink {
  icon: JSX.Element | null;
  value: string;
  label: string;
}

function getLinks(isAdmin: boolean): DrawerLink[] {
  return isAdmin
    ? [
        {
          icon: <User className="h-6 w-6" />,
          value: "/users",
          label: "Mentors",
        },
        {
          icon: <GraduationCap className="h-6 w-6" />,
          value: "/students",
          label: "Students",
        },
        {
          icon: <Home className="h-6 w-6" />,
          value: "/chapters",
          label: "Chapters",
        },
      ]
    : [
        {
          icon: null,
          value: "/roster",
          label: "Roster",
        },
        {
          icon: null,
          value: "/mentees",
          label: "My Mentees",
        },
      ];
}

export function Drawer({ isAdmin, version }: Props) {
  const links = getLinks(isAdmin);

  return (
    <div className="drawer-side pt-16">
      <label htmlFor="drawer" className="drawer-overlay"></label>
      <ul className="menu h-full w-80 border-r border-primary bg-base-200 p-4 text-base-content">
        {links.map(({ icon, label, value }, index) => (
          <li key={index}>
            <NavLink
              to={value}
              className={({ isActive }) =>
                isActive
                  ? "active mb-2 justify-between rounded font-semibold"
                  : "mb-2 justify-between rounded font-semibold"
              }
            >
              <div className="flex items-center gap-4">
                {icon}
                {label}
              </div>
              <NavArrowRight className="h-6 w-6" />
            </NavLink>
          </li>
        ))}
      </ul>
      <div
        data-testid="version"
        className="absolute bottom-0 left-2 z-10 text-sm italic"
      >
        Version {version}
      </div>
    </div>
  );
}
