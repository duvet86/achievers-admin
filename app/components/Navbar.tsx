import type { AzureUserWebAppWithRole, Environment } from "~/services";

import { Link } from "@remix-run/react";

import { ProfileCircle, Menu, LogOut } from "iconoir-react";

interface Props {
  currentUser: AzureUserWebAppWithRole;
  environment: Environment;
  version: string;
}

export function Navbar({ currentUser, environment, version }: Props) {
  const showEnvBadge = environment === "local" || environment === "staging";

  return (
    <nav className="navbar absolute left-0 top-0 z-10 h-16 bg-primary text-primary-content shadow-md">
      <div className="flex-none lg:hidden">
        <label htmlFor="drawer" className="btn btn-square btn-ghost">
          <Menu className="inline-block h-6 w-6 stroke-current" />
        </label>
      </div>
      <div className="flex-1">
        <Link to="/" className="btn btn-ghost text-xl normal-case">
          <img
            className="mr-4 w-16 rounded"
            src="/images/logo.png"
            alt="Logo"
          />
          <span className="text-xl font-semibold tracking-tight">
            Achievers WA
          </span>
          {showEnvBadge && (
            <span className="badge badge-lg ml-2">{environment}</span>
          )}
        </Link>
      </div>

      <div className="dropdown dropdown-end hidden lg:block">
        <div tabIndex={0} className="flex cursor-pointer items-center gap-2">
          <div className="font-semibold">{currentUser.email}</div>
          <label className="btn btn-circle btn-ghost">
            <div className="flex w-10 content-center justify-center rounded-full">
              <ProfileCircle />
            </div>
          </label>
        </div>
        <ul
          tabIndex={0}
          className="menu dropdown-content w-52 rounded-box bg-base-100 p-2 shadow"
        >
          <li>
            <Link className="font-semibold" to="/logout">
              <LogOut className="mr-2 w-6" />
              Logout
            </Link>
          </li>
          <li className="menu-title">
            <hr />
            <div className="mt-2 text-xs italic">Version {version}</div>
          </li>
        </ul>
      </div>
    </nav>
  );
}
