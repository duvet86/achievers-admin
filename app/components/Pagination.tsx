import classNames from "classnames";

import { FastArrowLeft, FastArrowRight, MoreHoriz } from "iconoir-react";

interface Props {
  range: number[];
  currentPageNumber: number;
  totalPageCount: number;
}

export function Pagination({
  range,
  currentPageNumber,
  totalPageCount,
}: Props) {
  const isPreviousDisabled = currentPageNumber === 0;
  const isNextDisabled =
    currentPageNumber === totalPageCount - 1 || totalPageCount === 0;

  return (
    <div className="join mt-4">
      <button
        type="submit"
        name="previousBtn"
        value="previousBtn"
        className={classNames("btn btn-outline join-item", {
          "btn-disabled": isPreviousDisabled,
        })}
        disabled={isPreviousDisabled}
        title="previous"
      >
        <FastArrowLeft />
      </button>
      {range.map((pageNumber, index) =>
        pageNumber === -1 ? (
          <button
            key={index}
            className="btn btn-disabled btn-outline join-item"
          >
            <MoreHoriz />
          </button>
        ) : (
          <button
            key={index}
            type="submit"
            name="pageNumberBtn"
            value={pageNumber - 1}
            className={classNames("btn btn-outline join-item", {
              "btn-active": currentPageNumber === pageNumber - 1,
            })}
          >
            {pageNumber}
          </button>
        ),
      )}
      <button
        type="submit"
        name="nextBtn"
        value="nextBtn"
        className={classNames("btn btn-outline join-item", {
          "btn-disabled": isNextDisabled,
        })}
        disabled={isNextDisabled}
        title="next"
      >
        <FastArrowRight />
      </button>
    </div>
  );
}
