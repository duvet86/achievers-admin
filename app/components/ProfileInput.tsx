import { useRef, useState } from "react";

import { Cancel } from "iconoir-react";

interface Props {
  defaultValue: string | null;
}

export default function ProfileInput({ defaultValue }: Props) {
  const inputRef = useRef<HTMLInputElement | null>(null);

  const [profilePicturePath, setProfilePicturePath] = useState<string | null>(
    defaultValue,
  );
  const [selectedImage, setSelectedImage] = useState<File | null>(null);

  const hasImage = selectedImage || profilePicturePath;

  return (
    <div className="card card-side m-8 bg-base-100 shadow-xl">
      <figure>
        <img
          src={
            selectedImage
              ? URL.createObjectURL(selectedImage)
              : profilePicturePath
              ? profilePicturePath
              : "/images/profile-picture.webp"
          }
          alt="profile"
          className="h-44 w-44"
        />
      </figure>
      <div className="card-body">
        <h2 className="card-title mb-4">Profile picture</h2>

        {hasImage && (
          <button
            className="btn btn-error w-32 gap-2"
            onClick={(e) => {
              e.preventDefault();
              if (inputRef.current) {
                inputRef.current.files = null;
                inputRef.current.value = "";
              }
              setSelectedImage(null);
              setProfilePicturePath(null);
            }}
          >
            <Cancel className="h-6 w-6" />
            Remove
          </button>
        )}

        <input
          type="file"
          name="profilePicture"
          className={hasImage ? "hidden" : "file-input w-full max-w-xs"}
          ref={inputRef}
          accept="image/png, image/gif, image/jpeg"
          onChange={(event) => {
            setSelectedImage(event.target.files?.[0] ?? null);
          }}
        />

        <input
          type="hidden"
          name="deleteProfilePicture"
          value={(!hasImage).toString()}
        />
      </div>
    </div>
  );
}
