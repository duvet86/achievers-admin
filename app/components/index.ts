export * from "./checkbox/Checkbox";
export * from "./date-input/DateInput";
export * from "./input/Input";
export * from "./textarea/Textarea";
export * from "./select/Select";
export * from "./file-input/FileInput";
export * from "./submit-form-button/SubmitFormButton";
export * from "./radio/Radio";
export * from "./autocomplete/Autocomplete";

export * from "./title/Title";
export * from "./sub-title/SubTitle";
export * from "./text/Text";
export * from "./back-header/BackHeader";
export * from "./pagination/Pagination";

export * from "./loading-spinner/LoadingSpinner";
export * from "./navbar/Navbar";
export * from "./drawer/Drawer";
export * from "./body/Body";

export * from "./not-found/NotFound";
export * from "./forbidden/Forbidden";
export * from "./unexpected-error/UnexpectedError";

export * from "./editor/Editor";
