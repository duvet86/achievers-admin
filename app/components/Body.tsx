import type { AzureUserWebAppWithRole, Environment } from "~/services";

import { Outlet } from "@remix-run/react";

import { Drawer } from "./Drawer";
import { Navbar } from "./Navbar";

interface Props {
  isAdmin: boolean;
  hasCompletedVolunteerAgreement?: boolean;
  version: string;
  currentUser: AzureUserWebAppWithRole;
  environment: Environment;
}

export function Body({
  currentUser,
  hasCompletedVolunteerAgreement,
  isAdmin,
  version,
  environment,
}: Props) {
  const showDrawer = isAdmin || hasCompletedVolunteerAgreement;

  return (
    <div className="drawer lg:drawer-open">
      <input id="drawer" type="checkbox" className="drawer-toggle" />
      <div className="drawer-content flex flex-col">
        <Navbar currentUser={currentUser} environment={environment} />

        <main className="mt-16 flex h-full flex-col overflow-y-auto bg-white p-4">
          <Outlet />
        </main>
      </div>

      {showDrawer && <Drawer isAdmin={isAdmin} version={version} />}
    </div>
  );
}
