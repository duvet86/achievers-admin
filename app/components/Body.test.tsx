import "@testing-library/jest-dom";

import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";

import { Body } from "~/components";

describe("Body", () => {
  it("Body snapshot admin", async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <Body
          isAdmin
          version="1"
          environment="local"
          currentUser={{
            appRoleAssignments: [],
            displayName: "",
            givenName: "",
            id: "1",
            mail: "test@test.com",
            surname: "",
            userPrincipalName: "",
            email: "",
          }}
        />
      </MemoryRouter>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("Body snapshot mentor", async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <Body
          isAdmin
          version="1"
          environment="local"
          currentUser={{
            appRoleAssignments: [],
            displayName: "",
            givenName: "",
            id: "1",
            mail: "test@test.com",
            surname: "",
            userPrincipalName: "",
            email: "",
          }}
        />
      </MemoryRouter>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
