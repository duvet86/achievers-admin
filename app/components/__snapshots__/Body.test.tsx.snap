// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Body > Body snapshot admin 1`] = `
<body>
  <div>
    <div
      class="drawer lg:drawer-open"
    >
      <input
        class="drawer-toggle"
        id="drawer"
        type="checkbox"
      />
      <div
        class="drawer-content flex h-screen flex-col"
      >
        <nav
          class="navbar absolute left-0 top-0 z-10 h-16 bg-primary text-primary-content shadow-md"
        >
          <div
            class="flex-none lg:hidden"
          >
            <label
              class="btn btn-square btn-ghost"
              for="drawer"
            >
              <svg
                class="inline-block h-6 w-6 stroke-current"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 5h18M3 12h18M3 19h18"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </label>
          </div>
          <div
            class="flex-1"
          >
            <a
              class="btn btn-ghost text-xl normal-case"
              href="/admin/home"
            >
              <img
                alt="Logo"
                class="mr-4 w-16 rounded"
                src="/images/logo.png"
              />
              <span
                class="text-xl font-semibold tracking-tight"
              >
                Achievers WA
              </span>
              <span
                class="badge badge-lg ml-2"
              >
                local
              </span>
            </a>
          </div>
          <div
            class="dropdown dropdown-end hidden lg:block"
          >
            <div
              class="flex cursor-pointer items-center gap-2"
              tabindex="0"
            >
              <div
                class="font-semibold"
              />
              <label
                class="btn btn-circle btn-ghost"
              >
                <div
                  class="flex w-10 content-center justify-center rounded-full"
                >
                  <svg
                    color="currentColor"
                    fill="none"
                    height="1.5em"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    width="1.5em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2z"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                    <path
                      d="M4.271 18.346S6.5 15.5 12 15.5s7.73 2.846 7.73 2.846M12 12a3 3 0 100-6 3 3 0 000 6z"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </div>
              </label>
            </div>
            <ul
              class="dropdown-content menu rounded-box w-52 bg-base-100 p-2 shadow"
              tabindex="0"
            >
              <li>
                <a
                  class="font-semibold"
                  href="/logout"
                >
                  <svg
                    class="mr-2 w-6"
                    color="currentColor"
                    fill="none"
                    height="1.5em"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    width="1.5em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 12h7m0 0l-3 3m3-3l-3-3M19 6V5a2 2 0 00-2-2H7a2 2 0 00-2 2v14a2 2 0 002 2h10a2 2 0 002-2v-1"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  Logout
                </a>
              </li>
            </ul>
          </div>
        </nav>
        <main
          class="mt-16 flex h-full flex-col overflow-y-auto p-4"
        />
      </div>
      <div
        class="drawer-side pt-16"
      >
        <label
          class="drawer-overlay"
          for="drawer"
        />
        <ul
          class="menu h-full w-80 border-r border-primary bg-base-200 p-4 text-base-content"
        >
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/home"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 8l9.732-4.866a.6.6 0 01.536 0L22 8M20 11v8a2 2 0 01-2 2H6a2 2 0 01-2-2v-8"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                Home
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/users"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 20v-1a7 7 0 017-7v0a7 7 0 017 7v1M12 12a4 4 0 100-8 4 4 0 000 8z"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                Mentors
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/students"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.573 8.463l8.659-4.329a.6.6 0 01.536 0l8.659 4.33a.6.6 0 010 1.073l-8.659 4.329a.6.6 0 01-.536 0l-8.659-4.33a.6.6 0 010-1.073z"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M22.5 13V9.5l-2-1M4.5 10.5v5.412a2 2 0 001.142 1.807l5 2.374a2 2 0 001.716 0l5-2.374a2 2 0 001.142-1.807V10.5"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                Students
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/chapters"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.485 3h-3.992l.5 5s1 1 2.5 1a3.23 3.23 0 002.139-.806.503.503 0 00.15-.465L21.076 3.5A.6.6 0 0020.485 3z"
                    stroke="currentColor"
                  />
                  <path
                    d="M16.493 3l.5 5s-1 1-2.5 1-2.5-1-2.5-1V3h4.5z"
                    stroke="currentColor"
                  />
                  <path
                    d="M11.993 3v5s-1 1-2.5 1-2.5-1-2.5-1l.5-5h4.5z"
                    stroke="currentColor"
                  />
                  <path
                    d="M7.493 3H3.502a.6.6 0 00-.592.501L2.205 7.73c-.029.172.02.349.15.465.328.29 1.061.806 2.138.806 1.5 0 2.5-1 2.5-1l.5-5z"
                    stroke="currentColor"
                  />
                  <path
                    d="M3 9v10a2 2 0 002 2h14a2 2 0 002-2V9"
                    stroke="currentColor"
                  />
                  <path
                    d="M14.833 21v-6a2 2 0 00-2-2h-2a2 2 0 00-2 2v6"
                    stroke="currentColor"
                    stroke-miterlimit="16"
                  />
                </svg>
                Chapters
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/roster"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 4V2m0 2v2m0-2h-4.5M3 10v9a2 2 0 002 2h14a2 2 0 002-2v-9H3zM3 10V6a2 2 0 012-2h2M7 2v4M21 10V6a2 2 0 00-2-2h-.5"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                Roster
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
        </ul>
        <div
          class="absolute bottom-0 left-2 z-10 text-sm italic"
          data-testid="version"
        >
          Version 
          1
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Body > Body snapshot mentor 1`] = `
<body>
  <div>
    <div
      class="drawer lg:drawer-open"
    >
      <input
        class="drawer-toggle"
        id="drawer"
        type="checkbox"
      />
      <div
        class="drawer-content flex h-screen flex-col"
      >
        <nav
          class="navbar absolute left-0 top-0 z-10 h-16 bg-primary text-primary-content shadow-md"
        >
          <div
            class="flex-none lg:hidden"
          >
            <label
              class="btn btn-square btn-ghost"
              for="drawer"
            >
              <svg
                class="inline-block h-6 w-6 stroke-current"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 5h18M3 12h18M3 19h18"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </label>
          </div>
          <div
            class="flex-1"
          >
            <a
              class="btn btn-ghost text-xl normal-case"
              href="/admin/home"
            >
              <img
                alt="Logo"
                class="mr-4 w-16 rounded"
                src="/images/logo.png"
              />
              <span
                class="text-xl font-semibold tracking-tight"
              >
                Achievers WA
              </span>
              <span
                class="badge badge-lg ml-2"
              >
                local
              </span>
            </a>
          </div>
          <div
            class="dropdown dropdown-end hidden lg:block"
          >
            <div
              class="flex cursor-pointer items-center gap-2"
              tabindex="0"
            >
              <div
                class="font-semibold"
              />
              <label
                class="btn btn-circle btn-ghost"
              >
                <div
                  class="flex w-10 content-center justify-center rounded-full"
                >
                  <svg
                    color="currentColor"
                    fill="none"
                    height="1.5em"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    width="1.5em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2z"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                    <path
                      d="M4.271 18.346S6.5 15.5 12 15.5s7.73 2.846 7.73 2.846M12 12a3 3 0 100-6 3 3 0 000 6z"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </div>
              </label>
            </div>
            <ul
              class="dropdown-content menu rounded-box w-52 bg-base-100 p-2 shadow"
              tabindex="0"
            >
              <li>
                <a
                  class="font-semibold"
                  href="/logout"
                >
                  <svg
                    class="mr-2 w-6"
                    color="currentColor"
                    fill="none"
                    height="1.5em"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    width="1.5em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 12h7m0 0l-3 3m3-3l-3-3M19 6V5a2 2 0 00-2-2H7a2 2 0 00-2 2v14a2 2 0 002 2h10a2 2 0 002-2v-1"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  Logout
                </a>
              </li>
            </ul>
          </div>
        </nav>
        <main
          class="mt-16 flex h-full flex-col overflow-y-auto p-4"
        />
      </div>
      <div
        class="drawer-side pt-16"
      >
        <label
          class="drawer-overlay"
          for="drawer"
        />
        <ul
          class="menu h-full w-80 border-r border-primary bg-base-200 p-4 text-base-content"
        >
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/home"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 8l9.732-4.866a.6.6 0 01.536 0L22 8M20 11v8a2 2 0 01-2 2H6a2 2 0 01-2-2v-8"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                Home
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/users"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 20v-1a7 7 0 017-7v0a7 7 0 017 7v1M12 12a4 4 0 100-8 4 4 0 000 8z"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                Mentors
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/students"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.573 8.463l8.659-4.329a.6.6 0 01.536 0l8.659 4.33a.6.6 0 010 1.073l-8.659 4.329a.6.6 0 01-.536 0l-8.659-4.33a.6.6 0 010-1.073z"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M22.5 13V9.5l-2-1M4.5 10.5v5.412a2 2 0 001.142 1.807l5 2.374a2 2 0 001.716 0l5-2.374a2 2 0 001.142-1.807V10.5"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                Students
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/chapters"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.485 3h-3.992l.5 5s1 1 2.5 1a3.23 3.23 0 002.139-.806.503.503 0 00.15-.465L21.076 3.5A.6.6 0 0020.485 3z"
                    stroke="currentColor"
                  />
                  <path
                    d="M16.493 3l.5 5s-1 1-2.5 1-2.5-1-2.5-1V3h4.5z"
                    stroke="currentColor"
                  />
                  <path
                    d="M11.993 3v5s-1 1-2.5 1-2.5-1-2.5-1l.5-5h4.5z"
                    stroke="currentColor"
                  />
                  <path
                    d="M7.493 3H3.502a.6.6 0 00-.592.501L2.205 7.73c-.029.172.02.349.15.465.328.29 1.061.806 2.138.806 1.5 0 2.5-1 2.5-1l.5-5z"
                    stroke="currentColor"
                  />
                  <path
                    d="M3 9v10a2 2 0 002 2h14a2 2 0 002-2V9"
                    stroke="currentColor"
                  />
                  <path
                    d="M14.833 21v-6a2 2 0 00-2-2h-2a2 2 0 00-2 2v6"
                    stroke="currentColor"
                    stroke-miterlimit="16"
                  />
                </svg>
                Chapters
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              class="mb-2 justify-between rounded font-semibold"
              href="/admin/roster"
            >
              <div
                class="flex items-center gap-4"
              >
                <svg
                  class="h-6 w-6"
                  color="currentColor"
                  fill="none"
                  height="1.5em"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  width="1.5em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 4V2m0 2v2m0-2h-4.5M3 10v9a2 2 0 002 2h14a2 2 0 002-2v-9H3zM3 10V6a2 2 0 012-2h2M7 2v4M21 10V6a2 2 0 00-2-2h-.5"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                Roster
              </div>
              <svg
                class="h-6 w-6"
                color="currentColor"
                fill="none"
                height="1.5em"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                width="1.5em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6l6 6-6 6"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </li>
        </ul>
        <div
          class="absolute bottom-0 left-2 z-10 text-sm italic"
          data-testid="version"
        >
          Version 
          1
        </div>
      </div>
    </div>
  </div>
</body>
`;
